### PR TITLE
Add primary language programming tutorials and related documents

### PR DESCRIPTION
**Pull request by texasdiaz**
Converting a few of the primary Wiki documents into ReadTheDocs. In this PR, converting and adding the following Programming Resources:

-- Tutorials --
Blocks Programming Tutorial
OnBot Java Programming Tutorial
Android Studio Programming Tutorial

-- General/Shared Supporting Docs --
FTC Control System Introduction
Required Materials
Using Your Android Device
Configuring your Android Devices
Connecting Devices to a Control or Expansion Hub
Configuring Your Hardware
Connecting a Laptop to the Program & Manage Network
Installing a Javascript Enabled Browser
Managing a Control Hub
Managing a Smartphone Robot Controller
Managing a Smartphone Driver Station

-- Language Specific Tutorial Docs --
Creating an Op Mode (Blocks, OnBot Java, Android Studio)
Running an Op Mode (Blocks, OnBot Java, Android Studio)
Controlling a Servo (Blocks,  OnBot Java, Android Studio)
Using Sensors (Blocks,  OnBot Java, Android Studio)
Blocks Reference Material
OnBot Java Reference Material
Downloading and Installing Android Studio
Downloading the FTC Android Studio Project

### CODE REVIEW

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:14]:
```
`<TEAM NUMBER>-DS`
`<TEAM NUMBER>`
```

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:42]:
```
`<TEAM NUMBER>-RC`
`<TEAM NUMBER>`
```

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:44]:
```
`<TEAM NUMBER>-DS`
`<TEAM NUMBER>`
```

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:29]: Use `.. note::` or `.. warning::`

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:309]: Use `.. note::` or `.. warning::`

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:4]: Should make header different from title of document

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:45]:
```
`DIRECT-`
```

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:47]:
```
`DIRECT-XK-9999-C-RC`
`ZU7if0hB`
```

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:56]:
```
`FTC-1Ybr`
`password`
```

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:80]:
```
`DIRECT-XK-9999-C-RC`
```

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:93]: This should not be a new line
```
`ZU7if0hB`
```

**Review by uvidyadharan — CHANGES_REQUESTED** [review]

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:93]: It's not a new line in the documentation. Text wrapping in this file is inconsequential.

File: docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst
Comment (cmacfarl) [reply docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst:16]: I think it is fine to capitalize the 'N' in Note in this usage.

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (cmacfarl) [reply docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:47]: I'm not convinced back ticks here significantly improve the readability of the text.  I'd probably leave this to the author's discretion as long as the author was consistent with usage.

File: docs/source/programming_resources/tutorial_specific/android_studio/enabling_developer_options/Enabling-Developer-Options.rst
Comment (cmacfarl) [reply docs/source/programming_resources/tutorial_specific/android_studio/enabling_developer_options/Enabling-Developer-Options.rst:9]: This could be **Important Note** The Control Hub ...

Which would make it consistent with the other Important Notes.

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (cmacfarl) [reply docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:29]: I'd advocate for consistent casing.

Comment (texasdiaz) [reply docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:29]: Yeah, I guess that begs the question of whether or not we're editing and revising all of Chris's old documentation or simply trying to convert it over.

File: docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst:134]: Browsers don't require it.

File: docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst]: done

File: docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst]: done

File: docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst]: I decided NOT to do this for all the "it will take <blah> to complete this" since it's not actually important.

File: docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst]: done

File: docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/managing_smartphone_rc/Managing-a-Smartphone-Robot-Controller.rst]: done

File: docs/source/programming_resources/tutorial_specific/android_studio/downloading_as_project_folder/Downloading-the-Android-Studio-Project-Folder.rst
Comment (texasdiaz) [reply docs/source/programming_resources/tutorial_specific/android_studio/downloading_as_project_folder/Downloading-the-Android-Studio-Project-Folder.rst:17]: done

File: docs/source/programming_resources/tutorial_specific/android_studio/enabling_developer_options/Enabling-Developer-Options.rst
Comment (texasdiaz) [reply docs/source/programming_resources/tutorial_specific/android_studio/enabling_developer_options/Enabling-Developer-Options.rst]: done

File: docs/source/programming_resources/tutorial_specific/android_studio/using_sensors/Using-Sensors-(Android-Studio).rst
Comment (texasdiaz) [reply docs/source/programming_resources/tutorial_specific/android_studio/using_sensors/Using-Sensors-(Android-Studio).rst]: Global search and replace.

File: docs/source/programming_resources/tutorial_specific/android_studio/using_sensors/Using-Sensors-(Android-Studio).rst
Comment (texasdiaz) [reply docs/source/programming_resources/tutorial_specific/android_studio/using_sensors/Using-Sensors-(Android-Studio).rst]: Global search and replace.

File: docs/source/programming_resources/tutorial_specific/android_studio/using_sensors/Using-Sensors-(Android-Studio).rst
Comment (texasdiaz) [reply docs/source/programming_resources/tutorial_specific/android_studio/using_sensors/Using-Sensors-(Android-Studio).rst]: Global search and replace.

File: docs/source/programming_resources/tutorial_specific/onbot_java/using_sensors/Using-Sensors-(OnBot-Java).rst
Comment (texasdiaz) [reply docs/source/programming_resources/tutorial_specific/onbot_java/using_sensors/Using-Sensors-(OnBot-Java).rst]: Global search and replace.

File: docs/source/programming_resources/tutorial_specific/onbot_java/using_sensors/Using-Sensors-(OnBot-Java).rst
Comment (texasdiaz) [reply docs/source/programming_resources/tutorial_specific/onbot_java/using_sensors/Using-Sensors-(OnBot-Java).rst]: Global search and replace.

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst]: done

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:309]: done

File: docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/configuring_android/Configuring-Your-Android-Devices.rst:44]: I think this goes beyond the scope of the conversion.

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:4]: It _is_ different.

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:45]: I think this goes beyond the scope of the conversion.

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:47]: I think this goes beyond the scope of the conversion.

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:56]: I think this goes beyond the scope of the conversion.

File: docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst
Comment (texasdiaz) [reply docs/source/programming_resources/shared/program_and_manage_network/Connecting-a-Laptop-to-the-Program-&-Manage-Network.rst:80]: I think this goes beyond the scope of the conversion.

**Comment by uvidyadharan** [comment]
I think "Programming Quick Links" on the main landing page should be sent to in-site tutorials instead of pdf versions.

**Comment by texasdiaz** [comment]
> I think "Programming Quick Links" on the main landing page should be sent to in-site tutorials instead of pdf versions.

Done.

File: docs/source/programming_resources/shared/control_system_intro/The-FTC-Control-System.rst
Comment (uvidyadharan) [inline docs/source/programming_resources/shared/control_system_intro/The-FTC-Control-System.rst:1]: This is a duplicate to the "[About the FIRST Tech Challenge](https://first-tech-challenge-ftcdocs--11.com.readthedocs.build/en/11/overview/ftccontrolsystem.html)" page seen at the top of the main TOC. In addition this conflicts with "[FTC Control System Overview](https://first-tech-challenge-ftcdocs--11.com.readthedocs.build/en/11/stubs/ftc_control_system_overview.htm)" page.

Comment (texasdiaz) [reply docs/source/programming_resources/shared/control_system_intro/The-FTC-Control-System.rst:1]: Yeah, we need a better "About FIRST Tech Challenge" page, I didn't replace it because I don't have anything better yet. I'll submit an issue for it.

Comment (texasdiaz) [reply docs/source/programming_resources/shared/control_system_intro/The-FTC-Control-System.rst:1]: Issue Created, https://github.com/FIRST-Tech-Challenge/ftcdocs/issues/12

Comment (texasdiaz) [reply docs/source/programming_resources/shared/control_system_intro/The-FTC-Control-System.rst:1]: Sorry, I totally didn't realize you pointed to the stub with respect to the conflict. I changed the main TOC to point to the real FTC Control System Overview page, and deleted the stub. Still need issue #12 .

**Comment by uvidyadharan** [comment]
PDF Build error due to .GIF files

File: docs/source/stubs/rookie_teams.rst
Comment (emcclurg) [inline docs/source/stubs/rookie_teams.rst:60]: Add the iFrames

**Review by emcclurg — APPROVED** [review]
iFrames are brilliant!  Much better than a PDF link or website link.

https://github.com/FIRST-Tech-Challenge/ftcdocs/blob/142a1c24b827ca9a9bf1ed058223889b886a4485/docs/source/stubs/game_manuals.rst

Once these are approved, add link within the persona pages (https://first-tech-challenge-ftcdocs--11.com.readthedocs.build/en/11/stubs/rookie_teams.html) with a Game Manual tab. 

There are smaller content edits to be made on Control System Intro and formatting edits to be worked through before kickoff.

File: docs/source/stubs/rookie_teams.rst
Comment (texasdiaz) [reply docs/source/stubs/rookie_teams.rst]: Done.

**Pull request merged**
